### PR TITLE
Update logger and set to level4

### PR DESCRIPTION
**Pull request by zdtsw**
## Description

- as part of ref: https://github.com/opendatahub-io/opendatahub-operator/issues/368
- but this wont suspend Warnings from kustomization.
- change wording for "Updating manifests..." to "Reconciling odh-dashboard from manifests:"
- update `secGenLog`
- change previous zap.Log from `deployment` true to false 
- rename function isSelfManaged to checkSelfManaged
- rename function isManagedRHODS to checkManagedRHODS

usages:
- can set log level in CSV as args to container
- change default level in main.go to build local `manager`

## new updated test result:
default (as loglevel 2)
```

{"level":"info","ts":"2023-08-04T16:14:01Z","msg":"Starting workers","controller":"datasciencecluster","controllerGroup":"datasciencecluster.opendatahub.io","controllerKind":"DataScienceCluster","worker count":1}
{"level":"info","ts":"2023-08-04T16:14:01Z","msg":"Starting workers","controller":"SecretGenerator","controllerGroup":"","controllerKind":"Secret","worker count":1}
# Warning: 'bases' is deprecated. Please use 'resources' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
# Warning: 'bases' is deprecated. Please use 'resources' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
# Warning: 'vars' is deprecated. Please use 'replacements' instead. [EXPERIMENTAL] Run 'kustomize edit fix' to update your Kustomization automatically.
```
set to loglevel 0
```
{"level":"info","ts":"2023-08-04T16:01:01Z","msg":"Starting workers","controller":"SecretGenerator","controllerGroup":"","controllerKind":"Secret","worker count":1}
{"level":"info","ts":"2023-08-04T16:01:01Z","msg":"Starting workers","controller":"datasciencecluster","controllerGroup":"datasciencecluster.opendatahub.io","controllerKind":"DataScienceCluster","worker count":1}
{"level":"info","ts":"2023-08-04T16:01:01Z","logger":"controllers.DataScienceCluster","msg":"Reconciling DataScienceCluster resources","Request.Namespace":"","Request.Name":"default"}
{"level":"info","ts":"2023-08-04T16:01:01Z","msg":"Starting workers","controller":"dscinitialization","controllerGroup":"dscinitialization.opendatahub.io","controllerKind":"DSCInitialization","worker count":1}
{"level":"info","ts":"2023-08-04T16:01:01Z","logger":"controllers.DSCInitialization","msg":"Reconciling DSCInitialization.","DSCInitialization":"","Request.Name":"default"}
{"level":"info","ts":"2023-08-04T16:01:01Z","logger":"controllers.DataScienceCluster","msg":"Reconcile odh-dashboard"}
{"level":"info","ts":"2023-08-04T16:01:04Z","logger":"controllers.DataScienceCluster","msg":"Reconciling odh-dashboard from manifests: /opt/manifests/odh-dashboard/base"}
# Warning: 'bases' is deprecated. Please use 'resources' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
```

can prints more info , e.g 
`Reconciling odh-dashboard from manifests: /opt/manifests/odh-dashboard/base"}`
which are called by `r.Log.Info()`

as for error, they are the same_
`{"level":"error","ts":"2023-08-04T16:14:43Z","logger":"controllers.DataScienceCluster","msg":"failed to reconcile codeflare on DataScienceCluster","instance.Name":"default","error":"no matches for kind \"InstaScale\" in version \"codeflare.codeflare.dev/v1alpha1\"","stacktrace":"github.com/opendatahub-io/opendatahub-operator/v2/controllers/datasciencecluster.`

## Updated test result:
**When set to logLevel=2 with least printout**
Screenshot from 2023-08-02 19-09-53

**When set to logLevel=0 with Info printout**
Screenshot from 2023-08-02 19-11-11

**When set to logLevel=1, previous r.Log.Info() change to debug level**
Screenshot from 2023-08-02 19-12-11

## How Has This Been Tested?

After 
Screenshot from 2023-08-01 18-00-04

^INFO of DSCI DSC are suspended.  Updating manifests as INFO

Before 
Screenshot from 2023-07-25 17-00-31

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: bundle/manifests/opendatahub-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -418,6 +418,7 @@ spec:
                 - --health-probe-bind-address=:8081
                 - --metrics-bind-address=127.0.0.1:8080
                 - --leader-elect
+                - --logLevel 4
                 command:
                 - /manager
                 image: quay.io/etirelli/opendatahub-operator:dev-0.0.6
```
Comment (VaishnaviHire) [inline bundle/manifests/opendatahub-operator.clusterserviceversion.yaml:424]: Not sure when this was introduced , but this is causing CI test failures. This should be `REPLACE_IMAGE:latest`

Comment (zdtsw) [reply bundle/manifests/opendatahub-operator.clusterserviceversion.yaml:424]: yeah, i have a fix for this https://github.com/opendatahub-io/opendatahub-operator/pull/388

Comment (etirelli) [reply bundle/manifests/opendatahub-operator.clusterserviceversion.yaml]: Sorry, I missed that. Thanks for fixing.

**Review by etirelli — APPROVED** [review]

**Pull request closed without merging**